test_plans/oneapi_local_memory.asciidoc

```diff
@@ -0,0 +1,103 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for sycl extension oneapi local memory
+
+This is a test plan for the extension enabling the declaration of local memory
+objects at the kernel functor scope described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_local_memory.asciidoc[sycl_ext_oneapi_local_memory.asciidoc]
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+=== Feature test macro
+
+All of the tests should use `#ifdef SYCL_EXT_ONEAPI_LOCAL_MEMORY` so they can
+be skipped if feature is not supported.
+
+=== Type coverage
+
+Let `T` is the type of an object to be allocated within local memory space.
+
+Tests are performed using the following types as `T`:
+
+In regular mode:
+
+* `int`
+* `float`
+* custom trivially destructible type
+
+In full conformance mode:
+
+* `bool`
+* `char`
+* `signed char`
+* `unsigned char`
+* `short`
+* `unsigned short`
+* `int`
+* `unsigned int`
+* `long`
+* `unsigned long`
+* `long long`
+* `unsigned long long`
+* `float`
+* `double`
+* custom trivially destructible type
+
+== Tests
+
+Extension introduces the following functions:
+
+* `template <typename T, typename Group, typename... Args>
+   multi_ptr<T, Group::address_space>
+   group_local_memory(Group g, Args&&... args);` _(1)_
+
+* `template <typename T, typename Group>
+   multi_ptr<T, Group::address_space>
+   group_local_memory_for_overwrite(Group g);` _(2)_
+
+We should check that the allocating memory is actually available for all
+work-items in the work-group and the allocated objects are being created
+correctly.
+
+=== Test description
+
+==== Check initial value of allocated object and the type of returned value
+
+Define and submit a work-group data parallel SYCL kernel with work group size
+1.
+
+For the `group_local_memory(Group g, Args&&... args)` in kernel scope do the
+following:
```

Comment:
It would be interesting to test this API in the case when `T` is an array.  The spec isn't clear how `args` are interpreted in this case, though.  Can there be one `arg` for each element of the array?  Is each element of the array initialized with the entire `args` pack?

@Pennycook, can you clarify the spec here?

Reply:
I think the intent was that `args` wasn't allowed in the array case.

Looking at the definition of `make_unique` (see [cppreference](https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique)), though, I'm now confused because they don't allow array types at all.  @rolandschulz, do you have any opinion on what the correct fix is here?  I don't remember why we decided to allow array types.

Reply:
Decided offline that we will support aggregate initialization.